### DISCUSSION TITLE
Github pages' images not working

### DISCUSSION THREAD

**Discussion by amagalla1394** [community/community#23086] [Other Feature Feedback, Questions, & Ideas]
<p>I am still new to github pages overall and started playing with some items by creating an item through VScode CLI.<br>
I have read online that GitHub pages is case sensitive and even after double-checking everything and even uploading the images directly into the repo, the images still do not display.</p>
<p>any help?</p>
<p>URL: <a href="https://github.com/amagalla1394/odin-recipe-git-test" rel="noopener nofollow ugc">amagalla1394/odin-recipe-git-test: The Odin Project Recipes project (github.com)</a></p>
<p>the page itself is also up</p>

**Comment by Gisgar3** [comment]
<p>Hi there, welcome to the <strong>GitHub Support Community</strong>! The reason the images on your GitHub Pages site will not display is because you’re trying to call to the photos in a directory that does not exist. For example, in <code>lasagna.html</code>, you’re trying to call to <code>/images/lasagna.html</code>, which does not exist <em>within</em> the folder that <code>lasagna.html</code> is located in. In order to fix this, just create an <code>image</code> folder in the same directory that <code>lasagna.html</code> is located in, then put your image within that <code>image</code> folder.</p>
<p>Hope this helps!</p>

**Reply by rachelgjippy** [reply]
Thanks I was experiencing the same, but I solved it now.

**Reply by Coderkreet** [reply]
Thanks a lot

**Reply by IamOneal** [reply]
Am having the same issue my images is not displaying how to fix this

**Reply by sat-4499** [reply]
awesome it works!!

**Reply by alaina96** [reply]
Hey I know this is an old discussion, but I am having the same issues as OP and I've tried every suggestion in this thread. I can't figure out why the images won't load.

**Reply by Gisgar3** [reply]
@alaina96 Can you go more in depth about your project's file-structure and the errors you're receiving?

**Reply by alaina96** [reply]
[Live site](https://alaina96.github.io/qr/ ) 
[Respository](https://github.com/alaina96/qr.git)

I have uploaded two photos for this project and I have tried removing them from the image folder, creating a new image folder, and have tried adding . in front of the the /

**Reply by manel118** [reply]
please i need your help with this , i tried everything but still not showing the images .. pleas take a look , the bpreblem i guess is in script.js where is pass the object info to inner html in drawItems() function.

https://github.com/manel118/BeautyShop

**Reply by Abunnasarmomin** [reply]
Awesome i tried this. And it is working. Thanks for your Help

**Comment by amagalla1394** [comment]
<p>I get that putting the images in the same directory would work much easier but isn’t it better practice to add images to a folder/directory of its own?<br>
I currently have both items in different directories but what i’m confused about is how the path is wrong?</p>
<p>if I am calling an image from lasagna.html for example… wouldn’t the image source path be ./images/lasagna.png ? or even add odin-recipe-git-test before the path?</p>
<p>let me know if that makes sense or if i am missing something.</p>
<p>thank you</p>

**Comment by Gisgar3** [comment, marked as answer]
<p>You should be able to call back to the parent directory so you can pull the images from there. I’d say that’s good practice. Looking at your repository <em>now</em>, the reason the images won’t display is because there’s no period at the start of the specified path. Instead of:<br>
<code>&lt;img src="/images/lasagna.png" alt="lasagna"&gt;</code><br>
use<br>
<code>&lt;img src="./images/lasagna.png" alt="lasagna"&gt;</code></p>
<p>I tested it on my end and it fixed the issue. Hope this helps out!</p>

**Reply by ryzen322** [reply]
thanks for helping my problem but if you have nested folder like this img/photo/lasagna.jpg you can use double that every nested folder 
TLDR just put double dot if you want to access parent/folder/img.jpg

<img src="../images/photo/lasagna.png" alt="lasagna">

**Reply by hafizafiz** [reply]
Thank you

**Reply by SolomonOdunusi** [reply]
Thank you very much. This solved my icon issue

**Reply by ddart-pvcc** [reply]
Thank you!

**Reply by roobens971** [reply]
Thank you and Thank you very much !!!

**Reply by Dread92** [reply]
god bless you

**Reply by Jaythecoder00** [reply]
your the  got dude i was struggling with this for days on and off i had the ' ../ ', but for some reason it didn't work until I did your solution god bless. 🙏🏾

**Reply by TayAki79** [reply]
Thank you very much. But it only helped me halfway =D
If I put the dots before my image-files inside my html-file it worked perfectly. But somehow, it did't apply on the image-files I'm using in the css-file as background-images. Anyone an idea to this issue?

**Reply by CaptColdDev** [reply]
Thank you so much!!!! I am in tears. You saved me

**Reply by douiux** [reply]
Thank you very much

**Reply by Dlopez2792** [reply]
thank you for this

**Reply by iamLazyCode** [reply]
> <img src="./images/lasagna.png" alt="lasagna">

awesome but can anyone tell why these images are not showing in the mobile app..please help

**Reply by RiverConsulting0** [reply]
@TayAki79 Did you figure this out? I'm running into the same issue, my css file isn't being read. 

> Thank you very much. But it only helped me halfway =D If I put the dots before my image-files inside my html-file it worked perfectly. But somehow, it did't apply on the image-files I'm using in the css-file as background-images. Anyone an idea to this issue?

**Reply by EthanAnaya** [reply]
THANK YOU!!!!!!!!!!!!!!!!!!!!!

**Reply by varinder76946** [reply]
still doesn't work. i tried every combination

**Reply by Gisgar3** [reply]
@varinder76946 Remember that file names and their extensions are case-sensitive. Check your images in HTML and make sure the extensions are correct based on case-sensitivity.

**Reply by Reuben20** [reply]
Github pages expect you to have an `assets` folder inside `docs` folder like this `docs/assets/`  and add all your images there.

Github pages use jekyll so it's a requirement to have the `assets` folder for images. Also jekyll can load markdown content so if you select the `root` path it can load your readme.md instead of looking for an index.html

**Reply by DemonCyborg007** [reply]
Man you are a life saver bro!! Thanks mate it worked the dot( . )

**Reply by WDWaleed** [reply]
Thank you so much @Reuben20. I've run into this "image path" problem so many times. I had to specify absolute URLs for some of my projects because the relative URLs never worked and the frustrating thing about absolute URLs was that I couldn't work on my local repo the same way I would have liked to but this fixed that problem. 

Thanks again. You're a lifesaver

**Reply by dnahol** [reply]
This worked! Thank you so much!

**Reply by supunrathnayake007** [reply]
omg,  i wasted lot of time doing nonsense. its confusing when it works fine in local environment.
Thank you so much.

**Reply by codingwithkushal** [reply]
brother you are a live saver

**Reply by omrojas** [reply]
Oh man. That feels good.

**Reply by Tee6** [reply]
> You should be able to call back to the parent directory so you can pull the images from there. I’d say that’s good practice. Looking at your repository _now_, the reason the images won’t display is because there’s no period at the start of the specified path. Instead of: `<img src="/images/lasagna.png" alt="lasagna">` use `<img src="./images/lasagna.png" alt="lasagna">`
> 
> I tested it on my end and it fixed the issue. Hope this helps out!

Im still having Problems, does anyone know what im doing wrong? My Repo is: https://github.com/Tee6/HomePage and the relevant links are in src/stores/CVStore.tsx

**Reply by dfuchs13** [reply]
I have tried absolutely everything:

I have recently published my Portfolio website on Github pages and NONE of my pictures load. I have spent the last 3 days straight troubleshooting and making changes but NOTHING works!! Even the AI in the Dev Tools could not figure it out. The paths seem to be correct and the code works locally when I run it off of VS Code. I am totally stumped and have no idea what to do now.

Here is the published site: https://dfuchs13.github.io/Portfolio/
and my GitHub code: https://github.com/dfuchs13/Portfolio/tree/gh-pages
and: https://github.com/dfuchs13/Portfolio/tree/main

**Reply by TheEzzy7** [reply]
I just ran mine the same way. VS code works locally and i've checked what I know to check... I'm following an online course but there is no "if your pictures don't show up do this section." So I'm also at a bit of a standstill.

**Comment by amagalla1394** [comment]
<p>I appreciate your help.<br>
it helped me better understand paths through CLI interaction as well, so it was all around really helpful.</p>
<p>Thank you</p>

**Reply by randomswords** [reply]
You can also use the image's link from GitHub as the source.

**Comment by amagalla1394** [comment]
<p>i did however end up placing the images in the same directory too though. i might test it later the other way</p>

**Comment by ElenaOhana** [comment]
<p>It solved the same problem I had and it looks like a very logical solution. :brain: Thank you :handshake:</p>

**Reply by hafizafiz** [reply]
thank you 2

**Comment by Stoneysa** [comment]
For me what worked was adding master to the start of the path "master/assets/images/" Since my branch was configured as Master

**Reply by CatP98** [reply]
This one worked for me, thanks!

**Comment by bell-kevin** [comment]
I think I'm having a similar issue, but I can't get my picture to show up.

[https://bellkevin.me/](https://bellkevin.me/)

[https://github.com/bell-kevin/Kevin-Bell/issues/4](https://github.com/bell-kevin/Kevin-Bell/issues/4)

**Reply by iamnexxed** [reply]
Have been facing the same issue recently

**Reply by bell-kevin** [reply]
I fixed it!

**Reply by bell-kevin** [reply]
https://raw.githubusercontent.com/bell-kevin/Kevin-Bell/main/README.md

**Comment by Siddharthbadal** [comment]
same issue. Images are not showing up in readme. have tried everything. images can't open images in the image folder of any repo. it keeps loading.

**Comment by yongxiChuang** [comment]
I have changed the code `<img src="./images/_MG_3799.JPG" alt="My test image" width = 300>`
but why can't the image show up?
https://github.com/yongxiChuang/cc.github.io.git
https://yongxichuang.github.io/cc.github.io/

And I go to the website looking for the DevTools, it shows up that there's no image folder in the source.
Please help me...

**Reply by MarczanHub** [reply]
Try by removing './ ' before "images/" in image source. It actually worked for me.

**Comment by pankajpuri** [comment]
https://github.com/pankajpuri/portfolio/tree/main/public/assests/images

images are stored : public/assests/images/pankajpuri.jpg

https://github.com/pankajpuri/portfolio/blob/main/src/components/profileSection.jsx
in profileSection.jsx 
<img
              className="`img-fluid w-100` profileImageBorder"
              src="../assests/images/pankajpuri.jpg"
              alt="profile"
            ></img>

Hi there, I have got react app. I deployed with gh-pages but I am unable to fetch images. 
I would really appreciate if anyone can help. Thank you.

**Reply by krish2248** [reply]
hey,  i m facing the same issue, and i m very new to gh-pages and entire  git overall. help me. i checked format is png or svg, size is also less, path is also correct. everything is ok still not showing

**Comment by gitwendley** [comment]
On Sat, Jun 10, 2023 at 1:12 AM Jane ***@***.***> wrote:

> Github pages expect you to have an assets folder inside docs folder like
> this docs/assets/ and add all your images thereGithub pages use jekyll so
> it need to have the assets folder for images. Also jekyll can load
> markdown content so if you select the root path it can load your
> readme.md instead of looking for an index.html
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/23086#discussioncomment-6138403>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5EQBP6NZEF75AECI2YSAUDXKP65BANCNFSM57GVNGVQ>
> .
> You are receiving this because you are subscribed to this thread.
>
i always create a file names Image but sometimes it’s work and sometimes he
doesn’t work

**Comment by karthik4dev** [comment]
i can't see any images in any github pages . Is there any problem with my computer ? i tried with chrome and edge  ☹️

**Reply by WDWaleed** [reply]
It depends on how you're specifying urls and file paths and if you're using Vite. 
If you're using Vanilla JS, HTML and CSS, then specify absolute file paths like this. /images/the_img.png

The slash means that the path starts from the root directory so you need to keep the images folder at the top level and then fetch images from there. If you're using VS Code, then it's intellisense will automatically help you out a bit.

For Vite, the process is different. If you need help with that then let me know.

**Comment by submagr** [comment]
One thing I realized is that the files stored as `git-lfs` does not work with github pages (see [this issue](https://github.com/git-lfs/git-lfs/issues/1342)). I was storing my images and video files as git-lfs. 

[Untracking the files from git lfs and adding them as regular files](https://stackoverflow.com/a/41961459/5756943) worked for me.

**Comment by BigYajuu** [comment]
I have read the whole discussion and other relevant articles and I still can't get my image to work on `gh-pages`. I use Parcel as a bundler for my website and I dynamically append images with jQuery/TS.

My website is now live: [https://bigyajuu.github.io/my-portfolio/](https://bigyajuu.github.io/my-portfolio/) and the images just would not show. I opened Inspect Element and the first thing I saw was the errors from the console: `GET https://bigyajuu.github.io/img/works/<literally any images>.png 404 (Not Found)`. But the `img/` folder IS in my gh-pages branch.

Incidentally, I found these error images are using the wrong URLs. From the Network tab of F12 menu you can see all these images lack the root folder `my-portfolio/`, which is the name of my repo and also the root path for my public website.

**Reply by V-Rezende** [reply]
Hey Martin, did you fix the issue with your images?
Would you be happy to share how do fix it?
Thanks

**Comment by BigYajuu** [comment]
> Hey Martin, did you fix the issue with your images? Would you be happy to share how do fix it? Thanks

Ok. The problem I faced has something to do with the root of image paths.

When hosting on your gh-pages you need to make sure the root is `/` instead of `./`, depending on your directory.

Inspect your error image files via Network tab of F12. Select your images and check the URLs. If done correctly, your path should include the entire gh-pages URL as root. e.g. `https://someone.github.io/repo/`, not `https://someone.github.io/`.

Here is a duplicate on a StackOverflow query page I made: [https://stackoverflow.com/questions/77624516/gh-page-hosting-parcel-v2-built-project-contains-static-image-assets-that-would](https://stackoverflow.com/questions/77624516/gh-page-hosting-parcel-v2-built-project-contains-static-image-assets-that-would).

**Reply by BigYajuu** [reply]
And quite a hassle that is as the reason it took me so long to find out the solution was that I did not turn on `tsc watch` when I repeatedly edited my Typescript code, causing a lot of void attempts.

**Comment by youyoumu** [comment]
`/img/...` doesn't work.
`img/...` works.

**Reply by supriya101** [reply]
Yes, it's working. Thank You

**Comment by ruhail12** [comment]
Good post but I was wondering if you could write a litte more on this subject? I’d be very thankful if you could elaborate a little bit further. Appreciate it!	<a href="https://www.autoholowaniekatowice.pl/skup-samochodow/">skup samochodów dostawczych śląsk</a>

**Comment by prudenix** [comment]
![Images](https://github.com/community/community/assets/172465061/006ee8df-0d2a-4ede-a9b5-7b29dcb1c6fd)

## Next.js
Ensure to add a dot (.) before your starting slash and possibly add the images folder.

**Comment by Lacricia** [comment]
Hi, I've tried both with ./ and just / and nothing works.... Can you please tell me what I'm doing wrong? Here's my gig: 
https://github.com/Lacricia/Juice

**Comment by rachelgjippy** [comment]
Can you kindly check line 11, where you open a “dub” and ensure that you
close it..

**Comment by rachelgjippy** [comment]
Can you kindly check line 11, where you open a “div class” and ensure that
you close it..

**Reply by Lacricia** [reply]
I just checked and it seems to be closed

**Comment by diazdesandi** [comment]
If you have trouble figuring out which path your images are in, open the element inspector and edit the path manually. That should tell you which is the correct one.

**Comment by ImOothman** [comment]
So I was having the same issue and weirdly (or maybe not weirdly to those with experience, I am completely new to software development) having the repo name at the start of my image name was causing them not to display on the Github page. So for example my repo name is landing-page and I had the images saved as landing-page-****, I switched this to the end ****-landing-page and images displayed fine on Github page. Good information to learn I guess!

**Comment by dnesgurung** [comment]
I'm having the same problem. I have tried using both "/" and "./" but nothing works. My github:
https://github.com/dnesgurung/odin-recipes.git

**Reply by Gisgar3** [reply]
I just tested it through your GitHub Pages deployment on your repository. You're trying to access an image on a directory outside (behind) of your recipe page. I confirmed that using the correct format to access outside directories fixed the issue.

For example, on your `aloo-nimki.html` page, use ``../images/Aloo-Nimki.png`` to access the directory before your recipe folder and then access the images folder in that directory.

**Reply by dnesgurung** [reply]
Thank you so much!! It worked!

**Comment by manavpatil-dev0** [comment]
This is usually a path issue rather than a GitHub Pages deployment issue.

Since this is a project site, be careful with root-relative paths. For example, `/images/food.jpg` tells the browser to look from the root of the domain, not necessarily inside your `odin-recipe-git-test` repository path. Relative paths such as `./images/food.jpg` or `../images/food.jpg` may be what you need depending on where the HTML file lives. Relative paths are a common fix for broken assets on project Pages sites. :contentReference[oaicite:0]{index=0}

Also check the exact filename, extension, and capitalization. `Pizza.jpg`, `pizza.jpg`, and `pizza.jpeg` should be treated as different paths on GitHub Pages.

A quick debugging trick: open DevTools → Network, reload the page, and click the failed image request. If it returns 404, copy the requested image URL and compare that path character-for-character with the file location in your repo. That will usually reveal the problem immediately.

If this helped resolve your issue, consider marking this as the answer so others with the same GitHub Pages problem can find it easily.
